doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1522,144 +1507,122 @@ Managing the local context
 Controlling the proof flow
 ------------------------------
 
-.. tacn:: assert (@ident : @type)
-   :name: assert
-
-   This tactic applies to any goal. :n:`assert (H : U)` adds a new hypothesis
-   of name :n:`H` asserting :g:`U` to the current goal and opens a new subgoal
-   :g:`U` [2]_. The subgoal :g:`U` comes first in the list of subgoals remaining to
-   prove.
+.. tacn:: assert ( @ident : @type ) {? by @ltac_expr3 }
+          assert @one_term {? @as_ipat } {? by @ltac_expr3 }
+   :name: assert; _
 
-   .. exn:: Not a proposition or a type.
+   The first form adds a new hypothesis named :n:`@ident` of type :n:`@type`.
+   If :n:`by @ltac_expr3` is not provided, the tactic creates a new subgoal
+   to prove this hypothesis, which becomes the first in the list of subgoals.
```

Comment:
Looking at `assert True`, it appears the subgoal becomes the first one in the context.  From `assert (True /\ False) by split`, it appears the new hypothesis must be fully proven by the ltac_expr3.

I added this as the first paragraph for assert:

   Adds a new hypothesis to the current subgoal and a
   new subgoal to prove the hypothesis.  The new subgoal becomes
   the first in the list of subgoals.  Then, if :n:`@ltac_expr3`
   is specified, it applies that tactic to fully prove the new subgoal (and
   otherwise fails).

and revised the rest a bit.